subsys/bluetooth/host/mesh/net.c

```diff
@@ -104,30 +104,25 @@ static bool check_dup(struct net_buf_simple *data)
 	return false;
 }
 
-static u64_t msg_hash(struct net_buf_simple *pdu)
+static u64_t msg_hash(struct bt_mesh_net_rx *rx, struct net_buf_simple *pdu)
 {
-	u8_t *tpdu_last;
 	u64_t hash;
+	u32_t ivi;
 
-	/* Last byte of TransportPDU */
-	tpdu_last = net_buf_simple_tail(pdu) - (CTL(pdu->data) ? 8 : 4) - 1;
+	/* Three least significant bytes of IVI */
+	ivi = BT_MESH_NET_IVI_RX(rx) & 0xffffff;
+	memcpy(&hash, &ivi, 3);
```

Comment:
I just realized that this will only do the right thing on little endian architectures, and since we at least in theory try to be endianness independent I'll need to fix this.

Reply:
Should be fixed now